src/components/MovieLibrary.jsx

```diff
@@ -1 +1,71 @@
 // implement MovieLibrary component here
+import React from 'react';
+// import movies from './../data';
+import SearchBar from './SearchBar';
+import MovieList from './MovieList';
+import AddMovie from './AddMovie';
+
+class MovieLibrary extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      searchText: '',
+      bookmarkedOnly: false,
+      selectedGenre: '',
+      movies: this.props.movies,
+      filteredMovies: this.props.filteredMovies,
+    };
+
+    this.onSearchTextChange = this.onSearchTextChange.bind(this);
+    this.onBookmarkedChange = this.onBookmarkedChange.bind(this);
+    this.onSelectedGenreChange = this.onSelectedGenreChange.bind(this);
+    this.AddMoviefunc = this.AddMoviefunc.bind(this);
+    this.moviesFilter = this.moviesFilter.bind(this);
+  }
+
+  onSearchTextChange(event) {
+    this.setState({ searchText: event.target.value });
+  }
+  onBookmarkedChange(event) {
+    this.setState({ bookmarkedOnly: event.target.checked });
+  }
+  onSelectedGenreChange(event) {
+    this.setState({ selectedGenre: event.target.value });
+  }
+  AddMoviefunc(addMovie) {  // add a movie do propo movies after button been clicked
+    this.setState((state) => ({ movies: [...state.movies, addMovie] }));  // "push" addmovie movies
+  }
+
+  moviesFilter() { // create a filter of displayed movies based on search bar and options
+    // Need to add 3 if below? Have to be checked
+    const { searchText, selectedGenre } = this.state; // Porque tirar movies não passe os testes?
```

Comment:
Uma opção aqui é adicionar o movies e bookmarkedOnly pra não ter que repetir o this.state ali embaixo :)
```suggestion
    const { searchText, selectedGenre, movies, boomarkedOnly } = this.state;
```